extension.js

```diff
@@ -15,198 +15,220 @@
-	    strip_v.style = strip_h.style;
-	    strip_v.opacity = strip_h.opacity;
-	    strip_v.width = strip_h.height / 4;
-
-	    focus_up.visible = strip_h.visible && settings.get_boolean('focusmode');
-	    focus_up.style = 'background-color : ' + settings.get_string('color-focus');
-
-	    focus_down.visible = strip_h.visible && settings.get_boolean('focusmode');
-	    focus_down.style = 'background-color : ' + settings.get_string('color-focus');
-	}));
-
-	// load previous state
-	if (settings.get_boolean('enabled'))
-	    toggleStrip();
-
-	// synchronize hot key
-	Main.wm.addKeybinding('hotkey', settings,
-			      Meta.KeyBindingFlags.IGNORE_AUTOREPEAT,
-			      Shell.ActionMode.ALL,
-			      () => {
-				  toggleStrip();
-			      }
-			     );
-
-	// watch for monitor changes
-	monitor_change_signal_id = Main.layoutManager.connect('monitors-changed', () => {
-	    num_monitors = Main.layoutManager.monitors.length;
-	});
-	
+      this.icon.gicon = this.icon_on;
+      this.syncStrip(true);
+      this.pointerWatch = this.pointerWatcher.addWatch(
+        interval,
+        this.syncStrip.bind(this)
+      );
     }
-
-    disable() {
-        if (this._indicator) {
-	    this._indicator.destroy();
-	    this._indicator = null;
-        }
-
-	if (monitor_change_signal_id)
-	    Main.layoutManager.disconnect(monitor_change_signal_id);
-
-	Main.wm.removeKeybinding('hotkey');
-	setting_changed_signal_ids.forEach(id => settings.disconnect(id));
-	setting_changed_signal_ids = [];
-	settings = null;
+    this.strip_h.visible = !this.strip_h.visible;
+    this.strip_v.visible = this.strip_h.visible;
+    this.focus_up.visible = this.strip_h.visible;
+    this.focus_down.visible = this.strip_h.visible;
+    this._settings.set_boolean("enabled", this.strip_h.visible);
+  }
+
+  enable() {
+    // Load settings
+    this._settings = this.getSettings();
+
+    this.icon_on = Gio.icon_new_for_string(
+      `${this.path}/icons/readingstrip-on-symbolic.svg`
+    );
+    this.icon_off = Gio.icon_new_for_string(
+      `${this.path}/icons/readingstrip-off-symbolic.svg`
+    );
+
+    this.pointerWatcher = getPointerWatcher();
+    this.currentMonitor = Main.layoutManager.currentMonitor;
+    this.num_monitors = Main.layoutManager.monitors.length;
+    this.monitor_change_signal_id = 0;
+
+    this.pointerWatch = null;
+    this.setting_changed_signal_ids = [];
+
+    // add to top panel
+    this._indicator = new PanelMenu.Button(0.0, this.metadata.name, false);
+    this.icon = new St.Icon({
+      gicon: this.icon_off,
+      style_class: "system-status-icon",
+    });
+    this._indicator.add_child(this.icon);
+    this._indicator.menu.addAction(_("Show/Hide"), () => this.toggleStrip());
+    this._indicator.menu.addAction(
+      _("Settings..."),
+      () => this.openPreferences(),
+      "settings-symbolic"
+    );
+
+    Main.panel.addToStatusArea(this.metadata.uuid, this._indicator);
```

Comment:
It'd be good to split the UI building stuff into its own file/lib. This will make it easier to read.